addons/product/pricelist.py

```diff
@@ -453,23 +298,48 @@ def _check_margin(self, cr, uid, ids, context=None):
                 return False
         return True
 
+    def _compute_pricelist_name(self, cr, uid, ids, name, args, context=None):
+        for record in self.browse(cr, uid, ids, context=context):
+            if record.compute_price == 'fixed':
+                res = str(record.fixed_price) + " " + record.currency_id.name
+            elif record.compute_price == 'percentage':
+                res = str(record.percent_price) + " % discount "
+            else:
+                res = str(abs(record.price_discount)) + " % discount + " + str(record.price_surcharge) + " surcharge "
+        return res
+
+    def _price_list_item_name(self, cr, uid, ids, name, args, context=None):
+        res = {}
+        for record in self.browse(cr, uid, ids, context=context):
+            res[record.id] = {'name': '', 'price_name': ''}
+            if record.categ_id:
+                res[record.id]['name'] = " Category : " + record.categ_id.name
+            elif record.product_tmpl_id:
+                res[record.id]['name'] = record.product_tmpl_id.name
+            elif record.product_id:
+                product_name = self.pool['product.product'].name_get(cr, uid, [record.product_id.id], context=context)[0][1]
+                product_name = product_name.replace('[%s]' % record.product_id.code, '')
+                res[record.id]['name'] = product_name
+            else:
+                res[record.id]['name'] = "All Products"
+            res[record.id]['price_name'] = self._compute_pricelist_name(cr, uid, ids, name, args, context=context)
+        return res
+
     _columns = {
-        'name': fields.char('Rule Name', help="Explicit rule name for this pricelist line."),
-        'price_version_id': fields.many2one('product.pricelist.version', 'Price List Version', required=True, select=True, ondelete='cascade'),
+        'name': fields.function(_price_list_item_name, type="char", string='Name', store=True, multi='price_item', help="Explicit rule name for this pricelist line."),
```

Comment:
here name is store in db and name is depend on category's name and product template name. so need to define trigger (depend) function to update name when category's name and product template's name is changed.

means store should be dict like {'product.pricelist.item': (lambda self, cr, uid, ids, c={}: ids, ['categ_id', 'product_tmpl_id], 10), 'product.template': (_get_product_template, ['name'], 10), 'product.category': (_get_category, ['name'], 10)}

if do not want to store name field, I propose to override name_get method and use display_name field instead of name field.